src/module.ts

```diff
@@ -1,6 +1,66 @@
-import { AppPlugin } from '@grafana/data';
+import { AppPlugin, PluginExtensionPoints, RawTimeRange } from '@grafana/data';
+import { DataQuery } from '@grafana/schema';
+import { GrafanaPyroscopeDataQuery } from '@grafana/schema/dist/esm/raw/composable/grafanapyroscope/dataquery/x/GrafanaPyroscopeDataQuery_types.gen';
 import { AppPluginSettings } from '@shared/types/AppPluginSettings';
 
 import { App } from './app/App';
 
-export const plugin = new AppPlugin<AppPluginSettings>().setRootPage(App);
+type PluginExtensionExploreContext = {
+  targets: DataQuery[];
+  timeRange: RawTimeRange;
+};
+
+type URLParamsBuilderProps = {
+  pyroscopeQuery: GrafanaPyroscopeDataQuery;
+  timeRange?: RawTimeRange;
+};
+
+function buildURL(props: URLParamsBuilderProps) {
+  const { timeRange, pyroscopeQuery } = props;
+
+  let timeRangeParam = '';
+  let explorationType = 'all';
+
+  let serviceName = props.pyroscopeQuery.labelSelector?.match(/service_name="([^"]+)"/)?.[1];
+
+  if (serviceName) {
+    explorationType = 'labels';
+  }
+
+  const datasourceParam = `var-dataSource=${pyroscopeQuery.datasource?.uid}`;
+  const serviceNameParam = serviceName ? `&var-serviceName=${serviceName}` : '';
+  const profileTypeParam = `&var-profileMetricId=${pyroscopeQuery.profileTypeId}`;
+  const explorationTypeParam = `&explorationType=${explorationType}`;
+  if (timeRange) {
+    timeRangeParam = `&from=${timeRange.from}&to=${timeRange.to}`;
+  }
+
+  return `/a/grafana-pyroscope-app/profiles-explorer?${datasourceParam}${serviceNameParam}${profileTypeParam}${timeRangeParam}${explorationTypeParam}`;
+}
+
+export const plugin = new AppPlugin<AppPluginSettings>()
+  .addLink<PluginExtensionExploreContext>({
```

Comment:
Just a silly request: could we move all the code below to a different file?

By keeping the entry point `module.ts` neat, it's easier to understand and maintain (e.g. think that we will have to bring back the settings page in the future). It also pave the way for adding some focused unit tests at some point.